File: server/app/schemas/analytics.py
"""
Схеми для аналітики
"""
from datetime import date
from pydantic import BaseModel


class DashboardStats(BaseModel):
    """Статистика дашборду"""
    total_tasks: int
    completed_tasks: int
    in_progress_tasks: int
    overdue_tasks: int
    completion_rate: float
    average_cycle_time: int | None = None  # мілісекунди


class VelocityData(BaseModel):
    """Дані velocity"""
    week_number: int
    year: int
    completed_tasks: int
    start_date: date
    end_date: date


class CycleTimeData(BaseModel):
    """Дані cycle time по виконавцю"""
    user_id: str
    user_name: str
    average_cycle_time_hours: float
    tasks_completed: int

File: app/src/main/java/com/example/doshka/domain/model/Analytics.kt
package com.example.doshka.domain.model

import java.time.LocalDate

/**
 * Статистика дашборду
 */
data class DashboardStats(
    val totalTasks: Int,
    val completedTasks: Int,
    val inProgressTasks: Int,
    val overdueTasks: Int,
    val completionRate: Float, // відсоток завершених задач
    val averageCycleTime: Long? // середній час виконання в мілісекундах
)

/**
 * Швидкість команди (velocity)
 */
data class VelocityData(
    val weekNumber: Int,
    val year: Int,
    val completedTasks: Int,
    val startDate: LocalDate,
    val endDate: LocalDate
)

/**
 * Дані для Cumulative Flow Diagram
 */
data class CumulativeFlowData(
    val date: LocalDate,
    val columnCounts: Map<String, Int> // columnId -> count
)

/**
 * Cycle Time по виконавцю
 */
data class CycleTimeData(
    val userId: String,
    val userName: String,
    val averageCycleTimeHours: Float,
    val tasksCompleted: Int
)

/**
 * Дані для Heatmap активності
 */
data class ActivityHeatmapData(
    val userId: String,
    val userName: String,
    val dayOfWeek: Int, // 1 = понеділок, 7 = неділя
    val actionsCount: Int
)

/**
 * Дані для Burndown chart
 */
data class BurndownData(
    val date: LocalDate,
    val remainingTasks: Int,
    val idealRemaining: Int
)
